LandSandBoat/server
# [Lua, Fishing] Inside the belly convert to IF and refactor

**_I affirm:_**

## What does this pull request do?

Convert quest "Inside the Belly" to IF.
- Refactored cutscene onTrigger for fish recommendations to be easier to read, eliminating long if/else chain
- Changed cutscene variations to be based on fishing level instead of player main job level
- Factored out onTrigger/onTrade into local functions
- Changed how roll and sum are calculated, such that instead of the roll being divided by 10 to match the range of the fish reward chance values, the fish reward values are multiplied by 10 to match the roll, in the range of (1, 1000).  There is no mathematical difference in the outcome odds.

## Steps to test these changes

!addquest 4 22 -- the real gift
!delquest 4 26 -- inside the belly
Set fishing skill to 29
!setskill 48 29
Talk to zaldon, doesn't start the quest
!additem 13808 -- fisherman's tunica
Equip tunica, (+1 to fishing skill, for a total of 30)
Talk to Zaldon, quest should start
Talk to Zaldon again, should recommend 4 fish to catch
Trade Zaldon various fish, and receive correct rewards, at correct rates
